### PR TITLE
[dispatcher] Make font sizes consistent with bottom menu numberpicker

### PR DESCRIPTION
**Pull request by zwim**
In the bottom menu you can select fractional font sizes (e.g. 17.5). 
Now you can change the font size to the same values as in the bottom menu.

---
This change is [Reviewable](https://reviewable.io/reviews/koreader/koreader/7626)

### CODE REVIEW

File: frontend/dispatcher.lua [hunk 1]
```diff
@@ -418,10 +420,11 @@ function Dispatcher:addItem(caller, menu, location, settings, section)
                             width = math.floor(Screen:getWidth() * 0.6),
                             value = location[settings] ~= nil and location[settings][k] or settingsList[k].default or 0,
                             value_min = settingsList[k].min,
-                            value_step = 1,
+                            value_step = settingsList[k].step or 1,
+                            precision = settingsList[k].precision,
```
Comment (NiLuJe) [inline frontend/dispatcher.lua:424]: This probably `nil`s NumperPicker's default `precision` (`"%02d"`) if that is unset in `settingsList`, right?

File: frontend/dispatcher.lua [hunk 2]
```diff
@@ -418,10 +420,11 @@ function Dispatcher:addItem(caller, menu, location, settings, section)
...
                             value_hold_step = 5,
                             value_max = settingsList[k].max,
-                            default_value = 0,
+                            default_value = settingsList[k].default,
```
Comment (NiLuJe) [inline frontend/dispatcher.lua:427]: Ditto on the no fallback here

File: frontend/dispatcher.lua [hunk 3]
```diff
@@ -460,7 +463,8 @@ function Dispatcher:addItem(caller, menu, location, settings, section)
                             width = math.floor(Screen:getWidth() * 0.6),
                             value = location[settings] ~= nil and location[settings][k] or 0,
                             value_min = settingsList[k].min,
-                            value_step = 1,
+                            value_step = settingsList[k].step or 1,
+                            precision = settingsList[k].precision,
```
Comment (NiLuJe) [inline frontend/dispatcher.lua:467]: Same

File: frontend/dispatcher.lua (hunk 1 above)
Comment (zwim) [reply frontend/dispatcher.lua:424]: Yes I think so. But we could add an default precision "%02d" her eif you want.

File: frontend/dispatcher.lua (hunk 2 above)
Comment (zwim) [reply frontend/dispatcher.lua:427]: yes.

File: frontend/dispatcher.lua (hunk 3 above)
Comment (zwim) [reply frontend/dispatcher.lua:467]: yes

File: frontend/dispatcher.lua [hunk 4]
```diff
@@ -414,14 +415,19 @@ function Dispatcher:addItem(caller, menu, location, settings, section)
                     end,
                     callback = function(touchmenu_instance)
                         local SpinWidget = require("ui/widget/spinwidget")
+                        local requested_precision = settingsList[k].precision
+                        if settingsList[k].step and math.floor(settingsList[k].step) ~= settingsList[k].step then
+                            requested_precision = "%0.1f"
+                        end
```
Comment (NiLuJe) [inline frontend/dispatcher.lua:421]: That's slightly funkier and doesn't change the fact that it stomps on the class defaults if it's unset in `settingsList` (or if step is unset, or if step is an integer) :?

Comment (zwim) [reply frontend/dispatcher.lua:421]: Yes, the fact that we use the defaults in numberpicker doesn't change. 
But the need of entering a precision string in the settings is dropped, which makes the settingsList shorter.

Comment (NiLuJe) [reply frontend/dispatcher.lua:421]: AFAICT, that still breaks NumberPicker because you risk stomping its `precision` default with a `nil`?

**Pull request merged**